plugin/helpers_bots.go

```diff
@@ -9,6 +9,39 @@ import (
 	"github.com/pkg/errors"
 )
 
+type shouldProcessMessageOptions struct {
+	AllowSystemMessages bool
+	AllowBots           bool
+	FilterChannelIDs    []string
```

Comment:
I think we are not on the same page what these options do. To me the behaviour should be:
If `FilterChannelIDs` is set, return `true` if and only if the message in question is in `FilterChannelIDs` and the other conditions options are fulfilled. The same goes for `FilterUserIDs`.

Do you agree with my understanding?

Reply:
Hi @hanzei. But the vice versa is true as well right? If the message is not posted in the `FilterChannelIDs` we can return false then and there itself. wdyt?

Reply:
Yes, but only if `FilterChannelIDs` if not an empty list.

Reply:
Agree. This will lead to bugs for the consumer of the method. Let me make the change.